core/modules/filter/filter.pages.inc

```diff
@@ -210,23 +210,27 @@ function filter_format_editor_image_form($form, &$form_state, $format) {
       'file_validate_image_resolution' => array($max_dimensions),
     ),
     '#wrapper_attributes' => array(
-      'data-editor-image-toggle' => t('Image Upload'),
+      'data-editor-image-toggle' => t('Upload an image'),
     ),
     '#parents' => array('fid'),
     '#weight' => -10,
     '#access' => $element_supports_uploads && $upload_settings['status'] && user_access('upload editor images'),
   );
 
-  // Use a "textfield" rather than "url" to allow relative paths.
-  $form['image']['src'] = array(
-    '#title' => t('Image URL'),
```

Comment:
Replacing the `Image URL` with `Select from library` breaks embedding external images of any kind. Rather than replacing `Image URL`, it seems like this should be a 3rd option.

Reply:
I suppose perhaps because I never link to external images I forgot all about this possible use.

At an early stage in producing this PR I did try using a third option but I couldn't get the toggle action to work satisfactorily. The script does strange things. Is there a way we could allow a user to enter an external URL in the field and then recognise that it is external and then override further modification of it? Personally I like the simplicity of the two option approach.

Reply:
> is there a way we could allow a user to enter an external URL in the field and then recognise that it is external (...) ?

The insert link functionality in CKEditor allows external URLs and internal paths.

Reply:
@olafgrabienski thank you for this reminder. We addressed this in https://github.com/backdrop/backdrop-issues/issues/3083 only 3 months ago and I had forgotten!

Reply:
I have now added a small piece of code to make use of some that already exists, and thereby make an exception for external links to avoid modifying the supplied URL.